File: DSKO/11to20/13.java
import java.util.Stack;

class Solution {
    public int solution(int[][] board, int[] moves) {
        Stack<Integer> st = new Stack<>();
        int answer = 0;
        int board_row = board.length;
        int board_col = board[0].length;
        
        for(int fixed_col : moves){
            for(int i = 0 ; i < board_row ; i++){
                int comp = board[i][fixed_col-1];
                if(comp != 0){
                    if(!st.empty() && st.peek() == comp){
                        st.pop();
                        answer += 2;
                        board[i][fixed_col-1] = 0;
                        break;
                    }else{
                        st.push(comp);
                        board[i][fixed_col-1] = 0;
                        break;
                    }
                }
            }
        }
        return answer;
    }
}

File: DSKO/11to20/13.py
def solution(board, moves):
    pang_cnt = 0
    board_row = len(board)
    board_col = len(board[0])
    basket = [0]
    print(board_row, board_col)
    
    for pick_col in moves:
        for row in range(board_row):
            if(board[row][pick_col-1] != 0):
                if(basket[-1] == board[row][pick_col-1]):
                    basket.pop()
                    pang_cnt += 2
                    board[row][pick_col-1] = 0
                    break
                else:
                    basket.append(board[row][pick_col-1])
                    board[row][pick_col-1] = 0
                    break
                
    return pang_cnt